read-cache.c

```diff
@@ -1547,6 +1548,7 @@ int refresh_index(struct index_state *istate, unsigned int flags,
 
 		replace_index_entry(istate, i, new_entry);
 	}
+	enable_fscache(0);
```

Comment:
There are quite a few places where we already enable the FSCache, with more to come, I am sure.

The good thing is that `enable_fscache(0)` does not simply disable the FSCache, despite what the name says, but it heeds nested pairs to `enable_fscache()`: if `1` was passed two times, the FSCache will be disabled only after `0` was passed two times.

Furthermore, the big problem with the previous regression was that the FSCache had to be updated because files had been written in the meantime, which I *think* cannot happen here. Or can it?

Let's see. I'll try to think aloud here: are there circumstances where we read file meta-data (e.g. `preload_index()`), then files are written/modified/deleted, then the index is refreshed? If so, then we will most likely need to add explicit `flush_fscache()` calls like we did in [`write_entry()` in `entry.c`](https://github.com/git-for-windows/git/blob/v2.19.0.windows.1/entry.c#L370-L371).

So let's look at the calls to `enable_fscache()`:
- there is one in `cmd_add()` in `builtin/add.c`. This one is not even paired with an `enable_fscache(0)` call, as `git add` is not supposed to change worktree files.
- there is `checkout_paths()` in `builtin/checkout.c`. This is paired with a call disabling the FSCache (which clears it), so we should be fine.
- there is one in `cmd_status()` in `builtin/commit.c`. Same argument as with `cmd_add()`.
- `mark_complete_and_common_ref()` in `fetch-pack.c`. Should be safe, as nothing should be written within the pair enabling/disabling the FSCache.
- `preload_index()` in `preload-index.c`. Paired. Should not write any file. Should be safe.

And that's it! So we're good!